kernel/fablib/runlevel/0_infrastructure/aws_ssh_key/aws_ssh_key.go

```diff
@@ -0,0 +1,120 @@
+package aws_ssh_key
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/openziti/fablab/kernel/model"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/crypto/ssh"
+	"io/ioutil"
+	"path"
+)
+
+var KeyManager = awsKeyManager{}
+
+func Express() model.InfrastructureStage {
+	return KeyManager
+}
+
+type awsKeyManager struct {
+}
+
+func (l awsKeyManager) Bootstrap(m *model.Model) error {
+	bindings := model.GetBindings()
+
+	if !bindings.Has("credentials", "aws", "ssh_key_name") {
+		bindings.Put(bindings.Must("environment"), "credentials", "aws", "ssh_key_name")
+	}
+
+	if bindings.Has("credentials", "ssh", "key_path") {
+		return nil
+	}
```

Comment:
So this is the bit that allows me to use my existing key, with my existing `~/.fablab/bindings.yml`?

Reply:
This is just providing default values for key name and key path.

Reply:
Then I guess I'm not seeing where this infrastructure stage bails out if I've already got an `ssh_key_name` set in my `bindings.yml`.